sql/telemetry/client_histogram_probe_counts_v1/query.sql

```diff
@@ -0,0 +1,577 @@
+RETURNS ARRAY<STRUCT<key STRING, value FLOAT64>> AS (
+  -- Given a MAP with duplicate keys, de-duplicates by summing the values of duplicate keys
+  (
+    WITH summed_counts AS (
+      SELECT
+        STRUCT<key STRING, value FLOAT64>(e.key, SUM(e.value)) AS record
+      FROM
+        UNNEST(map) AS e
+      GROUP BY
+        e.key
+    )
+
+    SELECT
+       ARRAY_AGG(STRUCT<key STRING, value FLOAT64>(record.key, record.value))
+    FROM
+      summed_counts
+  )
+);
+
+CREATE TEMP FUNCTION udf_buckets_to_map (buckets ARRAY<STRING>)
+RETURNS ARRAY<STRUCT<key STRING, value FLOAT64>> AS (
+  -- Given an array of values, transform them into a histogram MAP
+  -- with the number of each key in the `buckets` array
+  (
+    SELECT
+       ARRAY_AGG(STRUCT<key STRING, value FLOAT64>(bucket, 1.0))
+    FROM
+      UNNEST(buckets) AS bucket
+  )
+);
+
+CREATE TEMP FUNCTION udf_fill_buckets(input_map ARRAY<STRUCT<key STRING, value FLOAT64>>, buckets ARRAY<STRING>)
+RETURNS ARRAY<STRUCT<key STRING, value FLOAT64>> AS (
+  -- Given a MAP `input_map`, fill in any missing keys with value `0.0`
+  (
+    WITH total_counts AS (
+      SELECT
+        key,
+        COALESCE(e.value, 0.0) AS value
+      FROM
+        UNNEST(buckets) as key
+      LEFT JOIN
+        UNNEST(input_map) AS e ON SAFE_CAST(key AS STRING) = e.key
+    )
+    
+    SELECT
+      ARRAY_AGG(STRUCT<key STRING, value FLOAT64>(SAFE_CAST(key AS STRING), value))
+    FROM
+      total_counts
+  )
+);
+
+WITH latest_versions AS (
+  SELECT channel, MAX(CAST(app_version AS INT64)) AS latest_version
+  FROM
+    (SELECT
+      normalized_channel AS channel,
+      SPLIT(application.version, '.')[OFFSET(0)] AS app_version,
+      COUNT(*)
+    FROM `moz-fx-data-shared-prod.telemetry.main`
+    WHERE DATE(submission_timestamp) > DATE_SUB(CURRENT_DATE(), INTERVAL 90 DAY)
+    AND normalized_channel IN ("nightly", "beta", "release")
+    GROUP BY 1, 2
+    HAVING COUNT(*) > 1000
+    ORDER BY 1, 2 DESC
+    limit 100)
+  GROUP BY 1),
+
+normalized_histograms AS
+  (SELECT
+      client_id,
+      os,
+      app_version,
+      app_build_id,
+      hist_aggs.channel,
+      bucket_range.first_bucket,
+      bucket_range.last_bucket,
+      bucket_range.num_buckets,
+      aggregate.metric as metric,
+      aggregate.metric_type AS metric_type,
+      aggregate.key AS key,
+      aggregate.agg_type as agg_type,
+      latest_version,
+      udf_normalized_sum(
+        udf_aggregate_map_sum(ARRAY_AGG(STRUCT<key_value ARRAY<STRUCT <key STRING, value INT64>>>(aggregate.value)))) AS aggregates
+  FROM
+      clients_daily_histogram_aggregates_v1 AS hist_aggs
+  CROSS JOIN
+      UNNEST(histogram_aggregates) AS aggregate
+  LEFT JOIN latest_versions
+  ON latest_versions.channel = hist_aggs.channel
+  WHERE ARRAY_LENGTH(value) > 0
+  AND ((hist_aggs.channel = 'release' AND CAST(app_version AS INT64) >= (latest_version - 2))
```

Comment:
Hmm, it seems to me we should also be limiting by `date` since that is how `clients_daily_*` is partitioned. We limit to 90 days for `latest_versions`, we could do that, or we could find the min date for the channel/versions.